internal/app/machined/pkg/controllers/runtime/security_state.go

```diff
@@ -92,6 +95,13 @@ func (ctrl *SecurityStateController) Run(ctx context.Context, r controller.Runti
 				secureBootState = true
 			}
 
+			defaultEntry, err := sdboot.ReadVariable(efivario.NewDefaultContext(), sdboot.LoaderEntryDefaultName)
+			if err == nil {
+				if strings.HasPrefix(defaultEntry, "Talos-") {
+					bootedWithUKI = true
+				}
+			}
```

Comment:
Is there by chance a better option like getting some info about the running kernel from the sysfs or procfs? None known to me, just an idea to detect this by the current state and not boot config (while the current solution is fine)

Reply:
ideal would have been to see if the StubInfo has been set https://www.freedesktop.org/software/systemd/man/latest/systemd-stub.html#StubInfo, but since when we do kexec and skip sd-boot completely the best is to see if the Default entry has been set, which the installer always does